src/Common/AsynchronousMetrics.cpp

```diff
@@ -926,6 +936,48 @@ void AsynchronousMetrics::update(TimePoint update_time)
             tryLogCurrentException(__PRETTY_FUNCTION__);
         }
     }
+
+    if (cgroupcpu_max)
+    {
+        try {
+            cgroupcpu_max->rewind();
+
+            uint64_t quota = 0;
+            uint64_t period = 0;
+
+            readText(quota, *cgroupcpu_max);
```

Comment:
It also possible to have "max" string here instead of a number according to https://docs.kernel.org/admin-guide/cgroup-v2.html
It would be okay just to throw and skip, but logging an exception, in that case, is excessive

Reply:
Good point. It seems like this problem also exists for memory limits. What do you suggest we report when the value is `max`?

Reply:
IMO The best would be to report the total host memory `getMemoryAmountOrZero()` and `std::thread::hardware_concurrency()`. And ofc `getMemoryAmountOrZero()` should just skip the limit if it is "max"